### PR TITLE
WIP: script demonstrating piecewise-constant definition of coefficient 

### PR DESCRIPTION
**Pull request by gdmcbain**
This is one way to do #89.

### CODE REVIEW

**Comment by gdmcbain** [comment]
It's kind of one-dimensional but done with MeshTri as I wasn't sure that the facilities of MeshLine were the same and I was ultimately interesting myself in two-dimensional problems.

File: docs/examples/piecewise.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@bilinear_form
+def conduction(u, du, v, dv, w):
+    return w.w * sum(du * dv)
+
+
+k = np.ones(basis.N) * lengths[1]
+k[mesh.elements_satisfying(lambda x, y: x < 0)] = lengths[0]
+L = asm(conduction, basis, w=basis0.interpolate(k))
```
Comment (kinnala) [inline docs/examples/piecewise.py]:
```suggestion
X, Y = basis.global_coordinates()
k = 3 * (X < 0) + 2 * (X > 0) 
L = asm(conduction, basis, w=k)
```
This should also work.

Comment (kinnala) [reply docs/examples/piecewise.py]: Note: I should've read your issue more carefully. This approach is not as general as you probably require if you eventually want to use Submeshes to define k.

**Comment by gdmcbain** [comment]
Maybe instead of https://github.com/kinnala/scikit-fem/blob/e52e0ee1fd4fe0c1cd81365bc9700c04eb58cfc0/docs/examples/insulated.py#L38
it would be nicer to specify something like
```python
thermal_conductivity = {'wire': 101., 'insulation': 11.}
```
and read in the `field_data` from `meshio.read` which here contains
```python
{'convection': array([3, 1]), 'wire': array([1, 2]), 'insulation': array([2, 2])}
```
and can be parsed with
```python
{k: i for k, (i, d) in mesh.field_data.items() if d == mesh.dim()}
```
to give the mapping
```python
{'wire': 1, 'insulation': 2}
```

**Comment by kinnala** [comment]
I'll check this out and merge it tomorrow while trying to make the part where you modify mesh.py nicer.

**Pull request merged**